tests/integration/credset_test.go

```diff
@@ -0,0 +1,516 @@
+				Expect(cs.Spec.Name).To(Equal(updatedFirstName))
+				Expect("test-credential").To(Equal(updatedFirstCredName))
+				Expect("new-credential").To(Equal(updatedSecondCredName))
+			})
+		})
+	})
+})
+
+var _ = Describe("CredentialSet delete", func() {
+	Context("when an existing CredentialSet is delete", func() {
+		It("should run porter credentials delete", func() {
+			By("creating an agent action", func() {
+				ctx := context.Background()
+				ns := createTestNamespace(ctx)
+				name := "cs-delete-" + ns
+				testSecret := "secret-value"
+
+				Log(fmt.Sprintf("create k8s secret '%s' for credset", name))
+				csSecret := NewTestSecret(name, testSecret)
+				csSecret.ObjectMeta.Namespace = ns
+				Expect(k8sClient.Create(ctx, csSecret)).Should(Succeed())
+
+				Log(fmt.Sprintf("create credential set '%s' for credset", name))
+				cs := NewTestCredSet(name)
+				cs.Spec.Name = "cs-delete-test"
+				cs.ObjectMeta.Namespace = ns
+				cred := porterv1.Credential{
+					Name: "test-credential",
+					Source: porterv1.CredentialSource{
+						Secret: name,
+					},
+				}
+				cs.Spec.Credentials = append(cs.Spec.Credentials, cred)
+				cs.Spec.Namespace = ns
+
+				Expect(k8sClient.Create(ctx, cs)).Should(Succeed())
+				Expect(waitForPorterCS(ctx, cs, "waiting for credential set to apply")).Should(Succeed())
+				validateCredSetConditions(cs)
+
+				Log("verify it's created")
+				jsonOut := runAgentAction(ctx, "create-check-credentials-list", ns, []string{"credentials", "list", "-n", ns, "-o", "json"})
+				firstName := gjson.Get(jsonOut, "0.name").String()
+				numCreds := gjson.Get(jsonOut, "#").Int()
+				firstCredName := gjson.Get(jsonOut, "0.credentials.0.name").String()
+				Expect(int64(1)).To(Equal(numCreds))
+				Expect(cs.Spec.Name).To(Equal(firstName))
+				Expect("test-credential").To(Equal(firstCredName))
+
+				Log("delete a credential set")
+				Expect(k8sClient.Delete(ctx, cs)).Should(Succeed())
+				Expect(waitForCredSetDeleted(ctx, cs)).Should(Succeed())
+
+				Log("verify credential set is gone from porter data store")
+				delJsonOut := runAgentAction(ctx, "delete-check-credentials-list", ns, []string{"credentials", "list", "-n", ns, "-o", "json"})
+				delNumCreds := gjson.Get(delJsonOut, "#").Int()
+				Expect(int64(0)).To(Equal(delNumCreds))
+
+				Log("verify credential set CRD is gone from k8s cluster")
+				nsName := apitypes.NamespacedName{Namespace: cs.Namespace, Name: cs.Name}
+				getCS := &porterv1.CredentialSet{}
+				Expect(k8sClient.Get(ctx, nsName, getCS)).ShouldNot(Succeed())
+			})
+		})
+	})
+})
+
+//NewTestCredSet minimal CredentialSet CRD for tests
+func NewTestCredSet(csName string) *porterv1.CredentialSet {
+	cs := &porterv1.CredentialSet{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "porter.sh/v1",
+			Kind:       "CredentialSet",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "porter-test-me-",
+		},
+		Spec: porterv1.CredentialSetSpec{
+			//TODO: get schema version from porter version?
```

Comment:
yeah you could totally read the schema version from get.porter.sh/porter/pkg/credentials.SchemaVersion. I just created a PR that moves this value around a bit (https://github.com/getporter/porter/pull/2052) but I like how this would simplify and reduce some of the hard coding in our test data.